lib/datadog/tracing/sampling/span/rule.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'datadog/tracing/sampling/span/ext'
+
+module Datadog
+  module Tracing
+    module Sampling
+      module Span
+        # Span sampling rule that applies a sampling rate if the span
+        # matches the provided {Matcher}.
+        # Additionally, a rate limiter is also applied.
+        #
+        # If a span does not conform to the matcher, no changes are made.
+        class Rule
+          attr_reader :matcher, :sample_rate, :rate_limit
+
+          # Creates a new span sampling rule.
+          #
+          # @param [Sampling::Span::Matcher] matcher whether this rule applies to a specific span
+          # @param [Float] sample_rate span sampling ratio, between 0.0 (0%) and 1.0 (100%).
+          # @param [Numeric] rate_limit maximum number of spans sampled per second. Negative numbers mean unlimited spans.
+          def initialize(
+            matcher,
+            sample_rate: Span::Ext::DEFAULT_SAMPLE_RATE,
+            rate_limit: Span::Ext::DEFAULT_MAX_PER_SECOND
+          )
+
+            @matcher = matcher
+            @sample_rate = sample_rate
+            @rate_limit = rate_limit
+
+            @sampler = Sampling::RateSampler.new
+            @sampler.sample_rate = sample_rate
```

Comment:
I think this can be simplified as

```suggestion
            @sampler = Sampling::RateSampler.new(sample_rate)
```

...or am I missing something? 🤔

Reply:
I added a comment explaining: it's an annoyance with how `RateSampler` validates initializer parameters.
Changing it would be a breaking change, so I added a not for a change for v2.0.